src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -360,6 +368,19 @@ void RealmCoordinator::do_get_realm(RealmConfig&& config, std::shared_ptr<Realm>
         realm->update_schema(std::move(*schema), config.schema_version, std::move(migration_function),
                              std::move(initialization_function));
     }
+
+#ifdef REALM_ENABLE_SYNC
+    // run subscription initializer if the SDK has instructed core to do so. The subscription callback will be run if:
+    // 1. this is the first time we are creating the realm file
+    // 2. the database was already created, but this is the first time we are opening the db and the flag
+    // rerun_on_open was set
+    if (subscription_function) {
+        const auto current_subscription = realm->get_latest_subscription_set();
+        if (current_subscription.version() == 0 || (rerun_on_open && !db_created)) {
```

Comment:
IIUC `open_db`, it just opens the database but not create the file (it's created in a previous step). So `db_created` should be true here (nit: `db_opened` it's probably a better name). If `db_created` is false we should never created new subscriptions.